tests/cpp/StreamTests.cpp

```diff
@@ -53,66 +56,66 @@ const char* STREAM_ID_ENV_VAR = "DEFLECT_ID";
 const char* STREAM_HOST_ENV_VAR = "DEFLECT_HOST";
 }
 
-BOOST_AUTO_TEST_CASE(testParameterizedConstructorWithValues)
+BOOST_AUTO_TEST_CASE(testConstructionWithNoServer)
 {
-    const deflect::Stream stream("mystream", "somehost");
-    BOOST_CHECK_EQUAL(stream.getId(), "mystream");
-    BOOST_CHECK_EQUAL(stream.getHost(), "somehost");
+    BOOST_CHECK_THROW(deflect::Stream("id", "localhost"), std::runtime_error);
 }
 
-BOOST_AUTO_TEST_CASE(testDefaultConstructorReadsEnvironmentVariables)
```

Comment:
It cannot be replicated if a server is used that does not run on port 1701, which shouldn't be hardcoded in a unit test. DEFLECT_PORT as an env variable can be the only solution then.